File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/updatable/UpdatableObject.java
package com.agrogames.islandsofwar.engine.abs.updatable;

import com.agrogames.islandsofwar.engine.abs.graphics.GraphicsAdder;
import com.agrogames.islandsofwar.engine.abs.bullet.BulletAdder;
import com.agrogames.islandsofwar.engine.abs.map.MapProvider;
import com.agrogames.islandsofwar.engine.abs.unit.IUnitAdder;

public interface UpdatableObject {
    void update(MapProvider provider, BulletAdder bulletAdder, IUnitAdder unitAdder, GraphicsAdder graphicsAdder, float deltaTime);
}

File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/updatable/UpdatableObject.kt
package com.agrogames.islandsofwar.engine.abs.updatable

import com.agrogames.islandsofwar.engine.abs.bullet.BulletAdder
import com.agrogames.islandsofwar.engine.abs.graphics.GraphicsAdder
import com.agrogames.islandsofwar.engine.abs.map.MapProvider
import com.agrogames.islandsofwar.engine.abs.unit.IUnitAdder

interface UpdatableObject {
    fun update(provider: MapProvider, bulletAdder: BulletAdder, unitAdder: IUnitAdder, graphicsAdder: GraphicsAdder, deltaTime: Float)
}